fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -2867,6 +2878,27 @@ ACTOR Future<Void> processTransactionStateRequestPart(TransactionStateResolveCon
 
 } // anonymous namespace
 
+ACTOR Future<EncryptionAtRestMode> getEncryptionAtRestMode(Reference<AsyncVar<ServerDBInfo> const> db) {
```

Comment:
Am not sure if CP needs to read the co-ordinated state to fetch EncryptionAtRest (ClusterController reads co-ordinated states). The code seems to be same as what TLog uses to get the EaR mode, however, TLog usecase is to ensure that TLog persistent state (encrypted or not) matches with the cluster configuration. 
CP being an stateless process can just read the database configuration (first batch read the database config and cache it, especially given the EaR mode is _sticky_)

Any reason to have this method?

Reply:
I can change it but is there a reason against doing it this way?

Reply:
In proposed scheme, CPs on recruitment is reaching out to CC to fetch encryptionModes, however, I think there is no need for this dependency; hence, we can avoid adding the code altogether? (if it would have been existing code, then it is matter of cleanup, but, here we may be adding TechDebt of cleanup on ourselves? )